error-reporting/api/ErrorReportingSample/ErrorReportingSample.cs

```diff
@@ -56,16 +56,31 @@ private static void report(Exception e)
             Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID"));
 
         // Add a service context to the report. For more details see:
-        // https://cloud.google.com/error-reporting/reference/rest/v1beta1/projects.events#ServiceContext
-        ServiceContext serviceContext = new ServiceContext()
+        // https://cloud.google.com/error-reporting/reference/rest/v1beta1/ServiceContext
+        var assemblyName = System.Reflection.Assembly.GetExecutingAssembly().GetName();
+        var serviceContext = new ServiceContext()
         {
-            Service = "myapp",
-            Version = "1.1",
+            Service = assemblyName.Name,
+            Version = assemblyName.Version.ToString(),
         };
-        ReportedErrorEvent errorEvent = new ReportedErrorEvent()
+        // Add an error context to the report. For more details see:
+        // https://cloud.google.com/error-reporting/reference/rest/v1beta1/ErrorContext
+        var stackTrace = new System.Diagnostics.StackTrace(e, true);
+        var stackFrame = stackTrace.GetFrame(stackTrace.FrameCount - 1);
+        var errorContext = new ErrorContext()
+        {
+            ReportLocation = new SourceLocation()
+            {
+                FilePath = stackFrame.GetFileName(),
+                LineNumber = stackFrame.GetFileLineNumber(),
+                FunctionName = stackFrame.GetMethod().Name,
+            },
+        };
```

Comment:
A couple of comments here:

This information is not always available as it depends on what debug information you have available (.pdb files). So you need to protect against it not being present.

It's better to use the bottom frame as that's the frame where the exception was actually thrown, the rest show how the exception was bubbled up.

See what we do in [Google.Cloud.Diagnostics](https://github.com/googleapis/google-cloud-dotnet/blob/8509295870d00b72bc70b8d3a63a10bcacd8dafd/apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/ErrorReporting/ErrorReportingContextExceptionLogger.cs#L127-L143) for reference.

Reply:
The code is refactored to reflect on the best practices.